### ISSUE TITLE
Training with handwrite frac data

### ISSUE THREAD

**Issue by Inosonnia** [harvardnlp/im2markup#14]
I reproduced the result of printed and handwrited equations. Nice Results.
Besides, I generated a 2k printed 320x80 fraction eqns (e.g., \frac{1}{2}+\frac{1}{2} = 1) as training and val data.
The training step seems fine, but the testing result for arbitrary input is same (e.g., \frac{1}{3}+\frac{1}{3} = \frac{2}{3}).
In this case I set -max_num_tokens 50.
I am wondering is there any restrictions on image's format (or shape)? thank you.

**Comment by da03** [comment]
Hmm so what's the training and validation perplexity? Besides, 2k might be too small.

**Comment by Inosonnia** [comment]
Thanks for your response.

For the training perplexity, it converges quickly (batchsize = 20 at model-200) and fluctuates around 2.
For the testing perplexity, it ranges from 2 to 10 (with mean 7), and it seems useless since the outputted string is same for all input.

2k may be too small. I'll enlarge the training set and test it again.

**Comment by da03** [comment]
Hmm and perplexity of 2 is bad, ideally we should see something close to 1 (such as 1.1). I think there are two possibilities: 1) maybe you need to tune learning rate a little bit or 2) maybe the scale of the image is not appropriate (you can compare the size of symbols with the size of symbols in the provided dataset to make sure that they are of similar sizes, since the CNN structure might be sensitive to this)
